src/scitokens/utils/keycache.py

```diff
@@ -126,13 +127,23 @@ def getkeyinfo(self, issuer, key_id=None, insecure=False):
             if int(row['next_update']) < time.time() and self._check_validity(row):
                 # Try to update the key, but if it doesn't work, just return the saved one
                 try:
+                    # Close the sqllite lock
+                    conn.commit()
+                    conn.close()
+
+                    # Get the public key, probably from a webserver
                     public_key, cache_timer = self._get_issuer_publickey(issuer, key_id, insecure)
+
+                    # Get the sqllite connection again
+                    conn = sqlite3.connect(self.cache_location)
+                    curs = conn.cursor()
                     self._addkeyinfo(curs, issuer, key_id, public_key, cache_timer)
                     conn.commit()
                     conn.close()
                     return public_key
                 except:
-                    print("Unable to get key triggered by next update")
+                    logger = logging.getLogger("scitokens")
+                    logger.warning("Unable to get key triggered by next update")
                     conn.close()
```

Comment:
what happens if the `conn.close()` is called when the connection was already closed?

Can a SQLite connection be utilized as a Python guard?  All these `conn.close` / `conn.commit` seem messy.